skyportal/facility_apis/nicer.py

```diff
@@ -0,0 +1,449 @@
+from astropy.time import Time
+from datetime import datetime, timedelta
+import re
+import requests
+import urllib
+
+from . import FollowUpAPI
+from baselayer.app.env import load_env
+from baselayer.log import make_log
+
+from ..utils import http
+
+env, cfg = load_env()
+
+
+# Submission URL
+LOGIN_URL = f"{cfg['app.heasarc_endpoint']}/ark/LOGIN"
+NICER_URL = f"{cfg['app.heasarc_endpoint']}/ark/nicertoo/form.html"
+
+log = make_log('facility_apis/nicer')
+
+CLEANR = re.compile('<.*?>|&([a-z0-9]+|#[0-9]{1,6}|#x[0-9a-f]{1,6});')
+
+
+class NICERRequest:
+
+    """A JSON structure for NICER requests."""
+
+    def __init__(self, request):
+        """Initialize NICER request.
+
+        Parameters
+        ----------
+        request : skyportal.models.FollowupRequest
+            The request to add to the queue and the SkyPortal database.
+
+        """
+
+        self.requestgroup = self._build_payload(request)
+
+    def _build_payload(self, request):
```

Comment:
This is mostly a matter of style, but I would put all of this into the `__init__` instead of having a separate function.